new nest projects can not be served due to a missing cwd value targets.build.options

### Current Behavior

When creating a new nestjs app the application with `nx g @nx/nest:application` the app can not start. 

```
D:\dev\monorepo\node_modules\.bin\nx.cmd run myapp:serve:development

 NX   Running target serve for project myapp and 1 task it depends on:

———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

> nx run myapp:build:development

> webpack-cli build --node-env=development

Entrypoint main =

ERROR in main
Module not found: Error: Can't resolve './src' in 'D:\dev\monorepo'
resolve './src' in 'D:\dev\monorepo'
  using description file: D:\dev\monorepo\package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
    using description file: D:\dev\monorepo\package.json (relative path: ./src)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
        D:\dev\monorepo\src doesn't exist
      .js
        Field 'browser' doesn't contain a valid alias configuration
        D:\dev\monorepo\src.js doesn't exist
      .json
        Field 'browser' doesn't contain a valid alias configuration
        D:\dev\monorepo\src.json doesn't exist
      .wasm
        Field 'browser' doesn't contain a valid alias configuration
        D:\dev\monorepo\src.wasm doesn't exist
      as directory
        D:\dev\monorepo\src doesn't exist

webpack 5.98.0 compiled with 1 error in 42 ms
Warning: command "webpack-cli build --node-env=development" exited with non-zero status code
———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 NX   Running target serve for project myapp and 1 task it depends on failed

Failed tasks:

- myapp:build:development

Hint: run the command with --verbose for more details.


Process finished with exit code 130
```

### Expected Behavior

```
D:\dev\monorepo\node_modules\.bin\nx.cmd run myapp:serve:development

 NX   Running target serve for project myapp and 1 task it depends on:

———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

> nx run myapp:build:development

> webpack-cli build --node-env=development

chunk (runtime: main) main.js (main) 2.71 KiB [entry] [rendered]
webpack compiled successfully (4b66bc359d80249c)

> nx run myapp:serve:development

Build option outputFileName not set for myapp. Using fallback value of dist\apps\backends\myapp\main.js.
NX Daemon is not running. Node process will not restart automatically after file changes.
> nx run myapp:build:development  [local cache]
> webpack-cli build --node-env=development
chunk (runtime: main) main.js (main) 2.71 KiB [entry] [rendered]
webpack compiled successfully (4b66bc359d80249c)
 NX   Successfully ran target build for project myapp
Nx read the output from the cache instead of running the command for 1 out of 1 tasks.
Debugger listening on ws://localhost:9229/e3cdb980-236a-4db7-8981-b9af541c2b19
Debugger listening on ws://localhost:9229/e3cdb980-236a-4db7-8981-b9af541c2b19
For help, see: https://nodejs.org/en/docs/inspector

[Nest] 39240  - 09.07.2025, 11:22:02     LOG [NestFactory] Starting Nest application...
[Nest] 39240  - 09.07.2025, 11:22:02     LOG [InstanceLoader] AppModule dependencies initialized +8ms
[Nest] 39240  - 09.07.2025, 11:22:02     LOG [RoutesResolver] AppController {/api}: +4ms
[Nest] 39240  - 09.07.2025, 11:22:02     LOG [RouterExplorer] Mapped {/api, GET} route +2ms
[Nest] 39240  - 09.07.2025, 11:22:02     LOG [NestApplication] Nest application successfully started +1ms
[Nest] 39240  - 09.07.2025, 11:22:02     LOG 🚀 Application is running on: http://localhost:3000/api


```

### GitHub Repo

_No response_

### Steps to Reproduce

1. create a new nestjs app
2. call serve


### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node           : 22.14.0
OS             : win32-x64
Native Target  : x86_64-windows
npm            : 11.2.0

nx (global)            : 21.2.1
nx                     : 21.2.2
@nx/js                 : 21.2.2
@nx/jest               : 21.2.2
@nx/eslint             : 21.2.2
@nx/workspace          : 21.2.2
@nx/cypress            : 21.2.2
@nx/devkit             : 21.2.2
@nx/eslint-plugin      : 21.2.2
@nx/module-federation  : 21.2.2
@nx/nest               : 21.2.2
@nx/node               : 21.2.2
@nx/plugin             : 21.2.2
@nx/react              : 21.2.2
@nx/storybook          : 21.2.2
@nx/vite               : 21.2.2
@nx/web                : 21.2.2
@nx/webpack            : 21.2.2
typescript             : 5.8.3
---------------------------------------
Registered Plugins:
@nx/storybook/plugin
@nx/eslint/plugin
@nx/cypress/plugin
@naxodev/gonx
@nx/webpack/plugin
---------------------------------------
Community plugins:
@naxodev/gonx : 1.1.0
---------------------------------------
Cache Usage: 176.66 KB / 20.00 GB
```

### Failure Logs

```shell

```

### Package Manager Version

_No response_

### Operating System

- [ ] macOS
- [ ] Linux
- [x] Windows
- [ ] Other (Please specify)

### Additional Information

The solution is to add the `cwd` in targets at `build.options`

--- a/packages/nest/src/generators/application/application.spec.ts
+++ b/packages/nest/src/generators/application/application.spec.ts
@@ -43,6 +43,7 @@ describe('application generator', () => {
                 "args": [
                   "--node-env=development",
                 ],
+                "cwd": "my-node-app"
               },
             },
             "executor": "nx:run-commands",

--- a/packages/nest/src/generators/application/application.spec.ts
+++ b/packages/nest/src/generators/application/application.spec.ts
@@ -43,7 +43,6 @@ describe('application generator', () => {
                 "args": [
                   "--node-env=development",
                 ],
-                "cwd": "my-node-app"
               },
             },
             "executor": "nx:run-commands",
@@ -52,6 +51,7 @@ describe('application generator', () => {
                 "--node-env=production",
               ],
               "command": "webpack-cli build",
+              "cwd": "my-node-app",
             },
           },
           "copy-workspace-modules": {
@@ -278,6 +278,7 @@ describe('application generator', () => {
                     "--node-env=production",
                   ],
                   "command": "webpack-cli build",
+                  "cwd": "myapp",
                 },
               },
               "copy-workspace-modules": {
@@ -479,6 +480,7 @@ describe('application generator', () => {
                   "--node-env=production",
                 ],
                 "command": "webpack-cli build",
+                "cwd": "myapp",
               },
             },
             "copy-workspace-modules": {

--- a/packages/node/src/generators/application/application.spec.ts
+++ b/packages/node/src/generators/application/application.spec.ts
@@ -1132,6 +1132,7 @@ describe('app', () => {
       expect(buildTarget.executor).toBe('nx:run-commands');
       expect(buildTarget.options.command).toBe('webpack-cli build');
       expect(buildTarget.options.args).toEqual(['--node-env=production']);
+      expect(buildTarget.options.cwd).toEqual(project.sourceRoot)
       expect(buildTarget.configurations.development.args).toEqual([
         '--node-env=development',
       ]);

--- a/packages/node/src/generators/application/application.spec.ts
+++ b/packages/node/src/generators/application/application.spec.ts
@@ -1132,7 +1132,7 @@ describe('app', () => {
       expect(buildTarget.executor).toBe('nx:run-commands');
       expect(buildTarget.options.command).toBe('webpack-cli build');
       expect(buildTarget.options.args).toEqual(['--node-env=production']);
-      expect(buildTarget.options.cwd).toEqual(project.sourceRoot);
+      expect(buildTarget.options.cwd).toEqual(project.root);
       expect(buildTarget.configurations.development.args).toEqual([
         '--node-env=development',
       ]);

--- a/packages/node/src/generators/application/application.spec.ts
+++ b/packages/node/src/generators/application/application.spec.ts
@@ -1132,7 +1132,7 @@ describe('app', () => {
       expect(buildTarget.executor).toBe('nx:run-commands');
       expect(buildTarget.options.command).toBe('webpack-cli build');
       expect(buildTarget.options.args).toEqual(['--node-env=production']);
-      expect(buildTarget.options.cwd).toEqual(project.sourceRoot)
+      expect(buildTarget.options.cwd).toEqual(project.sourceRoot);
       expect(buildTarget.configurations.development.args).toEqual([
         '--node-env=development',
       ]);

--- a/packages/node/src/generators/application/lib/create-project.ts
+++ b/packages/node/src/generators/application/lib/create-project.ts
@@ -40,7 +40,7 @@ export function addProject(
     } else if (options.isNest) {
       // If we are using Nest that has the webpack plugin we need to override the
       // build target so that node-env can be set to production or development so the serve target can be run in development mode
-      project.targets.build = getNestWebpackBuildConfig();
+      project.targets.build = getNestWebpackBuildConfig(project);
     }
   }
   project.targets = {

--- a/packages/node/src/generators/application/lib/create-targets.ts
+++ b/packages/node/src/generators/application/lib/create-targets.ts
@@ -114,7 +114,9 @@ export function getServeConfig(options: NormalizedSchema): TargetConfiguration {
   };
 }
 
-export function getNestWebpackBuildConfig(project: ProjectConfiguration): TargetConfiguration {
+export function getNestWebpackBuildConfig(
+  project: ProjectConfiguration
+): TargetConfiguration {
   return {
     executor: 'nx:run-commands',
     options: {

--- a/packages/node/src/generators/application/lib/create-targets.ts
+++ b/packages/node/src/generators/application/lib/create-targets.ts
@@ -114,12 +114,13 @@ export function getServeConfig(options: NormalizedSchema): TargetConfiguration {
   };
 }
 
-export function getNestWebpackBuildConfig(): TargetConfiguration {
+export function getNestWebpackBuildConfig(project: ProjectConfiguration): TargetConfiguration {
   return {
     executor: 'nx:run-commands',
     options: {
       command: 'webpack-cli build',
       args: ['--node-env=production'],
+      cwd: project.sourceRoot,
     },
     configurations: {
       development: {

--- a/packages/node/src/generators/application/lib/create-targets.ts
+++ b/packages/node/src/generators/application/lib/create-targets.ts
@@ -122,7 +122,7 @@ export function getNestWebpackBuildConfig(
     options: {
       command: 'webpack-cli build',
       args: ['--node-env=production'],
-      cwd: project.sourceRoot,
+      cwd: project.root,
     },
     configurations: {
       development: {

